firefox_ui_tests/remote/security/test_safe_browsing_notification.py

```diff
@@ -0,0 +1,126 @@
+            FirefoxTestCase.tearDown(self)
+
+    def test_notification_bar(self):
+        with self.marionette.using_context('content'):
+            for item in self.test_data:
+                label = item['button_label']
+                report_page, unsafe_page = item['report_page'], item['unsafe_page']
+
+                # Navigate to the unsafe page
+                # Check "ignore warning" link then notification bar's "not badware" button
+                self.marionette.navigate(unsafe_page)
+                self.check_ignore_warning_button(unsafe_page)
+                self.check_not_badware_button(label, report_page)
+
+                # Return to the unsafe page
+                # Check "ignore warning" link then notification bar's "get me out" button
+                self.marionette.navigate(unsafe_page)
+                self.check_ignore_warning_button(unsafe_page)
+                self.check_get_me_out_of_here_button()
+
+                # Return to the unsafe page
+                # Check "ignore warning" link then notification bar's "X" button
+                self.marionette.navigate(unsafe_page)
+                self.check_ignore_warning_button(unsafe_page)
+                self.check_x_button()
+
+    def check_ignore_warning_button(self, unsafe_page):
+        button = self.marionette.find_element(By.ID, 'ignoreWarningButton')
+
+        # This isn't clickable by the time we get here and needs a
+        # pretty significant delay.
+        time.sleep(1)
+        button.click()
+
+        self.wait_for_condition(expected.element_present(By.ID, 'main-feature'))
+        self.assertRaises(NoSuchElementException, self.marionette.find_element,
+                          By.ID, 'ignoreWarningButton')
+        self.assertEquals(self.marionette.get_url(), unsafe_page)
+        # Clean up every time this function runs
+        self.perms.remove('www.itisatrap.org', 'safe-browsing')
+
+    # Check the not a forgery or attack button in the notification bar
+    def check_not_badware_button(self, button_label, report_page):
+        label = self.browser.get_property(button_label)
+        with self.marionette.using_context('chrome'):
+            tab_count = len(self.browser.tabbar.tabs)
+            button = (self.marionette.find_element(By.ID, 'content')
+                      .find_element('anon attribute', {'label': label}))
+            button.click()
+
+            self.wait_for_condition(lambda _: len(self.browser.tabbar.tabs) == tab_count + 1)
+            self.wait_for_condition(lambda _: report_page in self.browser.navbar.locationbar.value)
+            self.browser.tabbar.close_tab()
+
+    def check_get_me_out_of_here_button(self):
+        label = self.browser.get_property('safebrowsing.getMeOutOfHereButton.label')
+        with self.marionette.using_context('chrome'):
+            # is this the button in the notification bar, or the button on the page?
+            button = (self.marionette.find_element(By.ID, 'content')
```

Comment:
Lets call it notification panel for consistency and to not mix it up with any toolbar in the browser window.